### PR TITLE
reactor: detail header components

### PR DESCRIPTION
**Pull request by neurosnap**
Screenshot 2023-06-26 at 12 11 56 PM

### CODE REVIEW

File: src/ui/shared/tabs.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const Tabs = ({ tabs }: { tabs: TabItem[] }) => {
       {tabs.map((tab) => (
         <Tab
           key={tab.name}
-          isActive={tab.current}
+          isActive={!!tab.current}
```
Comment (madhuravius) [inline src/ui/shared/tabs.tsx:46]: nit - shouldn't need double bangs 🤔 (you told me this some time ago)

i prefer this sometimes, but we should be consistent about it

Comment (neurosnap) [reply src/ui/shared/tabs.tsx:46]: `tab.current` can now be undefined so it's not always a boolean.

File: src/ui/pages/stack-detail.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Navigate, useParams } from "react-router";
+
+import { stackDetailEnvsUrl } from "@app/routes";
+
+export const StackDetailPage = () => {
+  const { id = "" } = useParams();
+  return <Navigate to={stackDetailEnvsUrl(id)} replace />;
```
Comment (madhuravius) [inline src/ui/pages/stack-detail.tsx:7]: 🤔 until this is ready i'm guessing. we should probably do this more often

File: src/ui/shared/activity.tsx [hunk 3]
```diff
@@ -206,31 +206,29 @@ function ActivityTable({
...
+            "Status",
+            "Operation Type",
+            "Environment",
+            "User",
+            "Last Updated",
+            "Actions",
+          ]}
+        />
+      }
+      tableBody={
```
Comment (madhuravius) [inline src/ui/shared/activity.tsx:224]: table body should already have a parent div imo and this should contain the rows and be explicit it expects either:

`[]OpListRow` or `children` (ReactNode or something?) so this should just work OOB that way

**Review by madhuravius — APPROVED** [review]
LGTM, nearly exactly what i had in mind when planning on returning to this work

**Pull request merged**